pcdsdevices/gon.py

```diff
@@ -236,23 +255,348 @@ class Kappa(BaseInterface, Device):
+                logger.info("Starting now moving things!!!")
+                self.eta.mv(k_eta)
+                self.kappa.mv(k_kap)
+                self.phi.mv(k_phi)
+        except KeyboardInterrupt:
+            logger.info("Motion interrupted by ctrl+c")
+            self.stop()
+
+    @pseudo_position_argument
+    def forward(self, pseudo_pos):
+        """
+        Calculate a RealPosition from a given PseudoPosition.
+
+        Parameters
+        ----------
+        pseudo_pos : PseudoPosition
+            The pseudo position input.
+        Returns
+        -------
+        real_position : RealPosition
+            The real position output.
+        """
+        pseudo_pos = self.PseudoPosition(*pseudo_pos)
+        eta, kappa, phi = self.e_to_k(pseudo_pos.e_eta, pseudo_pos.e_chi,
+                                      pseudo_pos.e_phi)
+        return self.RealPosition(eta=eta, kappa=kappa, phi=phi,
+                                 x=self.x.position, y=self.y.position,
+                                 z=self.z.position)
+
+    @real_position_argument
+    def inverse(self, real_pos):
+        """
+        Calculate a PseudoPosition from a given RealPosition.
+
+        Parameters
+        ----------
+        real_position : RealPosition
+            The real position input.
+        Returns
+        -------
+        pseudo_pos : PseudoPosition
+            The pseudo position output.
+        """
+        real_pos = self.RealPosition(*real_pos)
+        e_eta, e_chi, e_phi = self.k_to_e(real_pos.eta, real_pos.kappa,
+                                          real_pos.phi)
+        return self.PseudoPosition(e_eta=e_eta, e_chi=e_chi, e_phi=e_phi)
+
+    def check_motor_step(self, eta, kappa, phi):
+        """
+        Check for the motor steps.
+
+        Compare desired movement destinations with current positions. If any of
+        the deltas are greater than their respective max step, ask the user for
+        confirmation.
+
+        Parameters
+        ----------
+        eta : number
+            Desired eta destination position.
+        kappa : number
+            Desired kappa destination position.
+        phi : number
+            Desired phi destination position.
+
+        Returns
+        -------
+        move_on : bool
+           `True` if motor step is smaller than the respective max step and/or
+           the user has confirmed yes.
+        """
+        eta_step = abs(eta - self.eta.position)
+        kappa_step = abs(kappa - self.kappa.position)
+        phi_step = abs(phi - self.phi.position)
+
+        is_eta_above_max = eta_step > self.eta_max_step
+        is_kappa_above_max = kappa_step > self.kappa_max_step
+        is_phi_above_max = phi_step > self.phi_max_step
+
+        if is_eta_above_max or is_kappa_above_max or is_phi_above_max:
+            d_str = '\nDo you really intend to do the following motions?\n'
+            t = PrettyTable(['Motor', 'Current position', 'to',
```

Comment:
Should I keep the `PrettyTable` here or should I implement what they had in the old code?
(there are pictures in the description of this PR of the differences)

Reply:
My 2c: PrettyTable looks nice

Reply:
ok thanks i'll leave that as is then